### PR TITLE
refactor(bitcoin_lib): use BCS for integer encoding and reader u32

### PR DESCRIPTION
**Pull request by vuvoth**
## Description

Closes: #246 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
  
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included doc comments for public functions
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary

## Summary by Sourcery

Adopt BCS-based integer encoding for Bitcoin library utilities and adjust readers and tests accordingly.

New Features:
- Introduce a helper to decode little-endian byte vectors into u32 values.

Enhancements:
- Replace manual bit-shifting logic with BCS serialization for u32 and u64 little-endian conversions and for VarInt payload encoding.
- Implement big-endian u64 byte conversion using reversed BCS output.
- Update reader u32 handling to use the new little-endian u32 decoder.

Tests:
- Add unit tests covering u32 reading from the byte reader, including edge cases.

### CODE REVIEW

**Comment by vuvoth** [comment]
@sourcery-ai title

File: bitcoin_lib/sources/encoding.move [hunk 1]
```diff
@@ -26,28 +26,21 @@ public fun le_bytes_to_u64(v: vector<u8>): u64 {
     number
 }
 
+/// Converts exactly 4 bytes in little-endian form to a u32 integer
```
Comment (sczembor) [inline bitcoin_lib/sources/encoding.move]:
```suggestion
/// Converts a 4-byte little-endian vector<u8> to a u32 integer
```

File: bitcoin_lib/sources/encoding.move [hunk 2]
```diff
@@ -26,28 +26,21 @@ public fun le_bytes_to_u64(v: vector<u8>): u64 {
...
+public fun le_bytes_to_u32(v: vector<u8>): u32 {
```
Comment (sczembor) [inline bitcoin_lib/sources/encoding.move]: lets add a test for this

**Review by sczembor — COMMENTED** [review]
lets add a test for the new function

File: bitcoin_lib/sources/encoding.move (hunk 2 above)
Comment (vuvoth) [reply bitcoin_lib/sources/encoding.move]: done!

**Review by sczembor — COMMENTED** [review]
pre-approve
